Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -797,4 +797,47 @@ def test_custom_command
         cmd(cmd, {"PATH" => "#{path}#{File::PATH_SEPARATOR}#{ENV["PATH"]}"})
     end
   end
+
+  def test_search
+    formula_file = CoreTap.new.formula_dir/"testball.rb"
+    formula_file.write <<-EOS.undent
+      class Testball < Formula
+        desc "Some test"
+        url "https://example.com/testball-0.1.tar.gz"
+      end
+    EOS
+
+    refute_predicate HOMEBREW_CACHE.join("desc_cache.json"),
+                     :exist?, "Cached file should not exist"
+
+    assert_match "testball", cmd("search")
+    assert_match "testball", cmd("search", "testball")
+    assert_match "testball", cmd("search", "homebrew\/homebrew-core\/testball")
+    assert_match "testball", cmd("search", "--desc", "Some test")
+
+    flags = Hash.new
+    flags["macports"] = "https://www.macports.org/ports.php?" +
+      "by=name&substr=testball"
+    flags["fink"] = "http://pdb.finkproject.org/pdb/browse.php?" +
+      "summary=testball"
+    flags["debian"] = "https://packages.debian.org/search?" +
+      "keywords=testball&searchon=names&suite=all&section=all"
+    flags["opensuse"] =
+      "https://software.opensuse.org/search?q=testball"
+    flags["fedora"] = "https://admin.fedoraproject.org" +
+      "/pkgdb/packages/%2Atestball%2A/"
+    flags["ubuntu"] = "http://packages.ubuntu.com/search?"+
+      "keywords=testball&searchon=names&suite=all&section=all"
+
+    flags.each do |key, value|
+      assert_equal "#{value}", cmd("search", "--#{key}",
```

Comment:
You can replace `"#{value}"` with `value` for simplicity. No need to use string interpolation here.